### PR TITLE
Updates to buildingdocs.rst

### PR DESCRIPTION
**Pull request by mstrout**
* Added in information about doxygen builds of compiler internal docs.
* Fixed a broken link, now is :ref:`Develop and test contributions locally`
* Using dashes instead of equal signs for all subheadings.
* Put in information about creating a new entry in the documentation sidebar

You can review this by doing a `make docs`, opening the generated html, and then navigating to Docs for Contributors » Best Practices for Contributors » Building Chapel Documentation.

### CODE REVIEW

File: doc/rst/developer/bestPractices/buildingdocs.rst [hunk 1]
```diff
@@ -84,8 +88,12 @@ Examples include:
...
 
 
-Linking between files
----------------------
+Linking within and between files
+--------------------------------
+
+Linking within the same file
+ - Create a label right before the target section heading, ``.. _section-label:``.
+ - Use ``:ref:`section-label``` elsewhere in the same document.
```
Comment (mppf) [inline doc/rst/developer/bestPractices/buildingdocs.rst:96]: AFAIK this strategy allows also linking between files - these section labels are global

File: doc/rst/developer/bestPractices/buildingdocs.rst [hunk 2]
```diff
@@ -193,6 +207,16 @@ heading the file is under.
      - Primers
      - doc/rst/meta/primers/index.rst + test/release/examples/primers/\*.chpl
 
+.. list-table:: **Use doxygen on .cpp files to generate html**
```
Comment (mppf) [inline doc/rst/developer/bestPractices/buildingdocs.rst:210]: actually we use doxygen and breathe -- maybe write it as "Use doxygen on .cpp files and then breathe and sphinx to generate html"

**Review by mppf — APPROVED** [review]

File: doc/rst/developer/bestPractices/buildingdocs.rst (hunk 1 above)
Comment (mstrout) [reply doc/rst/developer/bestPractices/buildingdocs.rst]: I simplified this a bit by not differentiating the within and between file pieces.

File: doc/rst/developer/bestPractices/buildingdocs.rst (hunk 2 above)
Comment (mstrout) [reply doc/rst/developer/bestPractices/buildingdocs.rst]: Done.

**Review by mppf — APPROVED** [review]

**Pull request merged**